src/com/apeshko/javacore/task_02/DynamicArray.java

```diff
@@ -0,0 +1,83 @@
+package com.apeshko.javacore.task_02;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public class DynamicArray {
+    private double[] items;
+    private int size;
+
+    public DynamicArray(int initialSize) {
+        size = 0;
+        items = new double[initialSize];
+    }
+
+    private void extend() {
+        double[] nextItems = new double[size * 2];
+
+        for (int i = 0; i < size; i += 1) {
+            nextItems[i] = items[i];
+        }
+
+        this.items = nextItems;
+    }
+
+    private void reduce() {
+        double[] nextItems = new double[(int) Math.floor(items.length / 2)];
+
+        for (int i = 0; i < size; i += 1) {
+            nextItems[i] = items[i];
+        }
+
+        this.items = nextItems;
+    }
+
+    public int add(double item) {
+        // extend array if needed
+        if (items.length == size) {
+            this.extend();
+        }
+
+        items[size++] = item;
+
+        return size - 1;
```

Comment:
typical contract for methods to add item to collection is return boolean, not quite clear what exactle returned from this method